src/puppetlabs/trapperkeeper/authorization/ring_middleware.clj

```diff
@@ -234,41 +234,61 @@
       request
       (ring-params/assoc-query-params request encoding))))
 
+(schema/defn add-rbac-subject
+  [request :- ring/Request
+   token->subject :- (schema/maybe IFn)]
+  (if token->subject
+    (if-let [token (get-in request [:headers "x-authentication"])]
+      (sling/try+
+        (assoc request :rbac-subject (token->subject token))
+        (catch [:kind :puppetlabs.rbac/token-revoked] {:keys [msg]}
+          (log/error "Failure validating RBAC token:" msg)
+          (assoc request :rbac-error msg))
+        (catch [:kind :puppetlabs.rbac/invalid-token] {:keys [msg]}
+          (log/error "Failure validating RBAC token:" msg)
+          (assoc request :rbac-error msg))
+        (catch [:kind :puppetlabs.rbac/token-expired] {:keys [msg]}
+          (log/error "Failure validating RBAC token:" msg)
+          (assoc request :rbac-error msg))
+        (catch [:kind :puppetlabs.rbac-client/connection-failure] {:keys [msg]}
+          (log/error "Failure validating RBAC token:" msg)
+          (assoc request :rbac-error msg)))
```

Comment:
Do you think this is worth trying to DRY up?

Reply:
ok gave it a shot

Reply:
This works for me. TIL about the `some->` macro :)